src/uwtools/drivers/driver.py

```diff
@@ -422,9 +424,13 @@ def _write_runscript(self, path: Path, envvars: Optional[dict[str, str]] = None)
         Write the runscript.
         """
         path.parent.mkdir(parents=True, exist_ok=True)
+        envvars = envvars or {}
+        threads = self._driver_config.get("execution", {}).get("threads")
+        if threads and "OMP_NUM_THREADS" not in envvars:
+            raise UWConfigError("Config specified threads but driver does not set OMP_NUM_THREADS")
```

Comment:
If `execution.threads` has been specified in the driver config but the driver does not provide support for it, raise an informative exception and decline to go further.